app/views/posts/index.html.erb

```diff
@@ -1 +1,24 @@
-<h1> Hello List of posts will appear right here</h1>
+<%= render 'shared/user_card', user: @user, user_card_class: 'user_card no_hover' %>
+
+<ul class='post_list'>
+  <% if @user.posts.empty? %>
+    <p class='user_empty-message'>No posts yet!</p>
+  <% else %>
+    <% @users.most_recent_five_posts.each do |post| %>
+      <li class='post_list_item'>
+        <%= render 'shared/post_card', post: post %>
+        <ul class='post_comment_list'>
+          <% if post.comments.empty? %>
+            <p class='comment_empty-message'>No comments yet!</p>
+          <% else %>
+            <% @posts.each do |comment| %>
+              <li class='post_comment_list_item'>
+                <%= render 'shared/comment_card', comment: comment %>
+              </li>
+            <% end %>
+          <% end %>
+        </ul>
```

Comment:
- [ ] Kindly pay attention to the instructions from the previous reviewer. In this post index page, you should render ALL the posts, not recent ones. There is also no method in your User model called `most_recent_five_posts`. But there is no need for recent posts on this page at all, to begin with.

  There is also no instance variable from the controller being passed on to this page that is `@users`, what you should have is `@user`. So kindly fix these issues on line 7. You can check that you are only rendering the posts for the `@user`. You may need to define `@user` within the index action of your controller.

  ```
  @user.posts.each do |post|
  ```

- [ ] On line 14 you are rendering all posts (based on your controller definition), but that should not be. The posts should only be the ones belonging to the `@user` and the comments that are rendered on line 14 should be `most_recent_five_comments`.

  ```
  post.most_recent_five_comments.each do |comment|
  ```

Reply:
Thank you for clearing that out @DammyShittu.